pdns/webserver.cc

```diff
@@ -145,6 +144,11 @@ void WebServer::apiWrapper(WebServer::HandlerFunction handler, HttpRequest* req,
     }
   }
 
+  if (!auth_ok && d_apikey.empty()) {
+    g_log<<Logger::Error<<req->logprefix<<"HTTP API Request \"" << req->url.path << "\": Authentication failed, API Key missing in config" << endl;
+    throw HttpUnauthorizedException("X-API-Key");
+  }
+
   if (!auth_ok) {
     g_log<<Logger::Error<<req->logprefix<<"HTTP Request \"" << req->url.path << "\": Authentication by API Key failed" << endl;
```

Comment:
A message talking about API Key failure is also given on password failure. Return code is also not relevant.